htap.go

```diff
@@ -22,5 +25,41 @@ func (qcc *queryContextCache) init() *queryContextCache {
 func (qcc *queryContextCache) add(qces ...queryContextEntry) {
 	qcc.mutex.Lock()
 	defer qcc.mutex.Unlock()
-	qcc.entries = append(qcc.entries, qces...)
+	var newQcesBasedOnPriorities []queryContextEntry
+	for _, newQce := range qces {
+		logger.Debugf("adding query context: %v", newQce)
+		newQceProcessed := false
+		for existingQceIdx, existingQce := range qcc.entries {
+			if newQce.ID == existingQce.ID {
+				newQceProcessed = true
+				if newQce.Timestamp > existingQce.Timestamp {
+					if newQce.Priority == existingQce.Priority {
```

Comment:
cannot we here just replace item in the list as we do in else branch?

Reply:
Actually we can.